app/src/main/java/com/sapuseven/untis/workers/NotificationSetupWorker.kt

```diff
@@ -115,8 +117,15 @@ class NotificationSetupWorker(context: Context, params: WorkerParameters) :
 					"preference_notifications_in_multiple"
 				).getValue()
 
-				val preparedItems = timetable.items.filter { !it.periodData.isCancelled() }
-					.sortedBy { it.startDateTime }.merged().zipWithNext()
+				val hiddenSubjects = applicationContext.stringDataStore(
+					user.id,
+					"timetable_hidden_elements"
+				).getValue()
+
+				val preparedItems = timetable.items.filter { !it.periodData.isCancelled() }.filter { item ->
+					!(item.periodData.subjects.isNotEmpty()
+						&& decodeMultipleStoredTimetableValues(hiddenSubjects)?.contains(item.periodData.subjects.first()) == true)
+				}.sortedBy { it.startDateTime }.merged().zipWithNext()
```

Comment:
This potentially ignores some subjects. The proposed solution should account for those cases, but I didn't test it yet.
```suggestion
				val hiddenSubjects = decodeMultipleStoredTimetableValues(
					applicationContext.stringDataStore(
						user.id,
						"timetable_hidden_elements"
					).getValue()
				).orEmpty()

				val preparedItems = timetable.items
					.filterNot { it.periodData.isCancelled() }
					.filterNot { item ->
						item.periodData.subjects.isNotEmpty() && hiddenSubjects.containsAll(item.periodData.subjects)
					}
					.sortedBy { it.startDateTime }
					.merged()
					.zipWithNext()
```